go.mod

```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/kueue
 
-go 1.22.4
+go 1.22.6
```

Comment:
Thank you for letting me know.
TBH, I would not like to select both approaches since the first never resolved the root issues for the environments depending on the `go list`, and the second is not recommended in the https://github.com/kubernetes/test-infra/pull/32983#issuecomment-2236655738.

So, I will try to seek another approach. But if I can not find another approach, we need to bump Go patch version in the gcb-docker-gcloud image.

Reply:
I did a little research and found that with toolchain not working well too. Every time on execute `go mod tidy` it is updating go version line and removing toolchain line on `go.mod` file. So we need to bump go version on `gcb-docker-gcloud`.

Another alternative is to add to env [here](https://github.com/kubernetes-sigs/kueue/blob/main/cloudbuild.yaml#L14).

```
GOTOOLCHAIN=go1.22.6
```

I tested this approach with `docker run --rm -e GOTOOLCHAIN=go1.22.6 -it --name test-integration -v .:/go/src/kubernetes-sigs/kueue -w /go/src/kubernetes-sigs/kueue gcr.io/k8s-staging-te
st-infra/gcb-docker-gcloud:v20240718-5ef92b5c36 -c "make yq"`. Looks like working fine.

Or we can try to use `.go-version` file (like [this](https://github.com/kubernetes/kubernetes/blob/master/.go-version)). But not sure it will work well.

Reply:
Before `go mod tidy`:

```
go 1.22.4

toolchain go1.22.6
```

After `go mod tidy`:
```
go 1.22.6
```

Reply:
try removing `.4` from the `go` directive.

Reply:
The same... :(

Reply:
maybe we just need to add that variable here?

https://github.com/kubernetes-sigs/kueue/blob/add6c3bb7c6a048fe347582a799718a92356627a/cloudbuild.yaml#L14

Reply:
The `LOCAL` seems to come from here https://github.com/docker-library/golang/blob/bb910a38695cd38c2b668671a8c19c32c6334db3/1.22/alpine3.20/Dockerfile#L121

We can probably just override it.

Reply:
> It should be GOTOOLCHAIN=auto. Not sure where this is set.

Yes, with `docker run --rm -e GOTOOLCHAIN=auto -it --name test-integration -v .:/go/src/kubernetes-sigs/kueue -w /go/src/kubernetes-sigs/kueue gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20240718-5ef92b5c36 -c "make yq"` working fine too.

Reply:
> The `LOCAL` seems to come from here https://github.com/docker-library/golang/blob/bb910a38695cd38c2b668671a8c19c32c6334db3/1.22/alpine3.20/Dockerfile#L121
> 
> We can probably just override it.

Good point!

Reply:
> maybe we just need to add that variable here?
> 
> https://github.com/kubernetes-sigs/kueue/blob/add6c3bb7c6a048fe347582a799718a92356627a/cloudbuild.yaml#L14

I think it should work.

Reply:
> > I did a little research and found that with toolchain not working well too. Every time on execute go mod tidy it is updating go version line and removing toolchain line on go.mod file.
> 
> How was the go.mod file looking before you run `go mod tidy`? IIUC, it should look like this:
> 
> ```
> go 1.22
> toolchain go1.22.6
> ```

Actually, this is my expectation since the Go starts to support the omitted patch version for the go module directive in the 1.22.4 or newer version: https://github.com/golang/go/issues/62278

But, indeed, the approach does not work well.